ras_party/controllers/queries.py

```diff
@@ -80,10 +80,17 @@ def update_respondent_details(respondent_data, respondent_id, session):
 
     logger.debug('Updating respondent details', respondent_id=respondent_id)
 
-    session.query(Respondent).filter(Respondent.party_uuid == respondent_id).update({
-                                     Respondent.first_name: respondent_data['firstName'],
-                                     Respondent.last_name: respondent_data['lastName'],
-                                     Respondent.telephone: respondent_data['telephone']})
+    if str(Respondent.first_name) != str(respondent_data['firstName']) or str(Respondent.last_name) != \
```

Comment:
Surely this will always evaluate to True as it's comparing to the class attributes (rather than an instance) of Respondent? Perhaps retrieve the instance (using the `party_uuid == respondent_id`) first then check for differences against that.